src/compiler/checker.ts

```diff
@@ -16089,6 +16089,21 @@ namespace ts {
             return false;
         }
 
+        function getJavaScriptClassType(symbol: Symbol): Type | undefined {
+            if (symbol && isDeclarationOfFunctionOrClassExpression(symbol)) {

```

Comment:
the caller just checked that symbol is defined, so `symbol &&` shouldn't be needed.